### PR TITLE
chore: upgrade to latest substyle

### PR DESCRIPTION
**Pull request by jfschwarz**
BREAKING CHANGE: requires upgrade to latest versions of substyle-jss/substyle-glamor if these are used

### CODE REVIEW

File: src/Stick.js [hunk 1]
```diff
@@ -22,28 +21,36 @@ import { type AlignT, type ApiPropsT, type PositionT } from './flowTypes'
...
+      position: 'absolute',
+      zIndex: 99,
+      textAlign: 'left',
+    },
+  },
+  getModifiers
+)
+
+function Stick(props: ApiPropsT) {
+  const style = useStyle((props: Object))
```
Comment (frontendphil) [inline src/Stick.js:36]: This cast shows that there is something wrong with how `useStyle` is typed. This shouldn't be necessary. I also find it a bit weird, how this new API now makes it essentially impossible to destructure the props parameter. 

I'd have expected a syntax like: 

```
const style = useStyle(modifiers)

// or

const style = useStyle(styles, modifiers)
```

Which would require much less rewriting. What is the benefit of the `createUseStyles` hooks creator method?

Comment (frontendphil) [reply src/Stick.js:36]: Does this have something to do with allowing style overwrites? Could this also be more explicit by including it in the API? 

```
const style = useStyle(styles, modifiers, props.style)
```

**Comment by jfschwarz** [comment]
@frontendphil looks like we'll have to update the `ModifiersT` flow type in substyle

**Comment by frontendphil** [comment]
@jfschwarz I wonder whether there will ever be a PR where you **just** did what you said you did. For instance, **only** upgrade `substyle` and not also change a ton of other things. Even if these are small this always makes PRs harder to review.

**Comment by frontendphil** [comment]
I'm gonna approve this PR if you also review mine ;) I've already created a `v3` branch so that I can pull changes onto the latest v3 release without forcing to upgrade substyle.

**Pull request merged**